### ISSUE TITLE
Random crashes when creating, cloning or changing the parent of a part.

### ISSUE THREAD

**Issue by nerdkoopa** [CapsAdmin/pac3#402]
Random crashes straight to desktop happen whenever creating parts, cloning parts, or changing the parent of a part. There doesn't seem to be a pattern to it, but some parts crash more often than others. Here's the crash dump from when changing the parent crashed it: https://www.dropbox.com/s/qouuwn68z32bzu3/hl2_170422_crash_2017_4_25T15_53_16C0.mdmp?dl=0
The way the addon is right now, it's completely unusable.

**Comment by AeoMayo** [comment]
Confirmed, having the same issue, seems like parts will suddenly gain an arbitrary hard number limit of parts that can be attached to it, ANY kind of part, and exceeding that number crashes straight to desktop with that same kind of dump. You can remove a part, and then replace it with something else, and it won't crash since that doesn't hit the weird limit, but as long as you add MORE than what there is already, you _will_ crash.

It seems to differ slightly from person to person. I've had some close friends test my outfits that have this issue to see if they crash as well when adding a part, some do, some don't, and some crash on things that I don't crash on when they add something to it. They also can try different workarounds that either do or do not work for me for some freaking reason.

It didn't seem like this issue happened at all until the last Gmod update.

**Comment by Cordon** [comment]
Yep, getting it after the newest update too, hasn't happened ever to me before the update.

**Comment by cerool** [comment]
It's been happening to Me too and it's been very annoying but I'm starting to think this bug has to deal with how good Your computer is because of how it varies. I'm really hoping this issue gets fixed soon though because it really make Me unmotivated to do anything PAC related now knowing that I'm going to crash every time I try and improve the thing I made.

**Comment by ToastyEngineer** [comment]
Literally has nothing to do with your computer.

**Comment by Count-Of-Pac** [comment]
This issue has been happening to me as well, and it's slowly getting worse. I can't even add proxies anymore without crashing to desktop instantly. And my game stutters whenever I clone any other part I might have, inevitably crashing after I add anything else a few minutes later. The only near-solution I've found is spastic saving, which seems to prevent crashing from cloning some things, or adding most new parts, but anything proxy or particle related seems to instantly kill my game if I have more than one  in the same group. I've cleared all of my addons, cleared all of my downloads in my gmod directory, reinstalled the game, verified game files, and still nothing. It's only getting worse, almost to the point that I can't do anything anymore.

**Comment by Earu** [comment]
Does it do that with a specific outfit or anything?

**Comment by Count-Of-Pac** [comment]
It does it with anything I try to make. if I add any individual parts on their own with nothing else, I'm fine; but it seems that once I add more than a few parts to my outfit I crash if I don't save spastically. And, if I have any parts under a parent proxy, and try to move those parts, no matter what it is or how many of them there are, I crash too. 

Proxies seem to crash me no matter what I do, other parts I can prevent crashing by saving every time I do anything in the editor.

**Comment by Vandarx** [comment]
This is a pretty frustrating issue that's been plaguing me as well. I haven't seen much mention of it outside of this thread, however; is this still a problem for everyone else? I've crashed 4 times in the last 10 minutes just by trying to get around it.

**Comment by CapsAdmin** [comment]
this seems to be fixed in the github version but still active in workshop version
